api/python/provisioner/commands/set_swupgrade_repo.py

```diff
@@ -333,6 +376,54 @@ def dynamic_validation(self, params: inputs.SWUpgradeRepo, targets: str):  # noq
 
         return repo.metadata
 
+    @staticmethod
+    def _setup_python_index(repo: inputs.SWUpgradeRepo):
+        """
+        Setup Python index
+
+        Parameters
+        ----------
+        repo: inputs.SWUpgradeRepo
+            SW upgrade repository parameters
+
+        Returns
+        -------
+
+        """
+        def setup_default():
+            _config = ConfigParser()
+
+            _config['global'] = {
+                'timeout': '60',
+                'index-url': f'file:///{python_index_path}'
+            }
+            return _config
+
+        iso_mount_dir = repo.target_build / repo.release
+        python_index_path = iso_mount_dir / CORTX_PYTHON_ISO_DIR
+        if python_index_path in iso_mount_dir.iterdir():
+            if not Path(PIP_CONFIG_FILE).exists():
+                config = setup_default()
+
+            else:
+                config = ConfigParser()
+                config.read(PIP_CONFIG_FILE)
+                extra_index_url = config['global'].get('extra-index-url', None)
+                if extra_index_url:
+                    if str(python_index_path) not in extra_index_url:
+                        # NOTE: if is needed to avoid index duplication
+                        config['global']['extra-index-url'] = (
```

Comment:
So it would be written like:
```
[global]
timeout = 60
index-url = aaa
extra-index-url = <previous value>
        <add value>
```
Is it ok for pip?

Reply:
Yes, it is ok for pip. Before `set_swupgrade_repo` pip.conf looked like
```ini
[global]
index-url = http://download.zope.org/simple
timeout = 60
trusted-host = download.zope.org
                        pypi.org
extra-index-url = http://pypi.org/simple
                           https://pypi.org/simple
```
After `set_swupgrade_repo` pip.conf looks like:
```
[global]
index-url = http://download.zope.org/simple
timeout = 60
trusted-host = download.zope.org
        pypi.org
extra-index-url = http://pypi.org/simple
        https://pypi.org/simple
        file:///opt/seagate/cortx/upgrades/2.0.0-757/python_deps
```
And pip works:
```
 pip3 download aiohttp --dest=/tmp/test/
Collecting aiohttp
  Using cached https://files.pythonhosted.org/packages/99/f5/90ede947a3ce2d6de1614799f5fea4e93c19b6520a59dc5d2f64123b032f/aiohttp-3.7.4.post0.tar.gz
  Saved /tmp/test/aiohttp-3.7.4.post0.tar.gz
Collecting attrs>=17.3.0 (from aiohttp)
  Using cached https://files.pythonhosted.org/packages/c3/aa/cb45262569fcc047bf070b5de61813724d6726db83259222cd7b4c79821a/attrs-20.3.0-py2.py3-none-any.whl
  Saved /tmp/test/attrs-20.3.0-py2.py3-none-any.whl
Collecting chardet<5.0,>=2.0 (from aiohttp)
  Using cached https://files.pythonhosted.org/packages/19/c7/fa589626997dd07bd87d9269342ccb74b1720384a4d739a1872bd84fbe68/chardet-4.0.0-py2.py3-none-any.whl
  Saved /tmp/test/chardet-4.0.0-py2.py3-none-any.whl
Collecting multidict<7.0,>=4.5 (from aiohttp)
  Using cached https://files.pythonhosted.org/packages/1c/74/e8b46156f37ca56d10d895d4e8595aa2b344cff3c1fb3629ec97a8656ccb/multidict-5.1.0.tar.gz
  Saved /tmp/test/multidict-5.1.0.tar.gz
Collecting async_timeout<4.0,>=3.0 (from aiohttp)
  Using cached https://files.pythonhosted.org/packages/e1/1e/5a4441be21b0726c4464f3f23c8b19628372f606755a9d2e46c187e65ec4/async_timeout-3.0.1-py3-none-any.whl
  Saved /tmp/test/async_timeout-3.0.1-py3-none-any.whl
Collecting yarl<2.0,>=1.0 (from aiohttp)
  Using cached https://files.pythonhosted.org/packages/97/e7/af7219a0fe240e8ef6bb555341a63c43045c21ab0392b4435e754b716fa1/yarl-1.6.3.tar.gz
  Saved /tmp/test/yarl-1.6.3.tar.gz
Collecting typing_extensions>=3.6.5 (from aiohttp)
  Using cached https://files.pythonhosted.org/packages/60/7a/e881b5abb54db0e6e671ab088d079c57ce54e8a01a3ca443f561ccadb37e/typing_extensions-3.7.4.3-py3-none-any.whl
  Saved /tmp/test/typing_extensions-3.7.4.3-py3-none-any.whl
Collecting idna-ssl>=1.0 (from aiohttp)
  Using cached https://files.pythonhosted.org/packages/46/03/07c4894aae38b0de52b52586b24bf189bb83e4ddabfe2e2c8f2419eec6f4/idna-ssl-1.1.0.tar.gz
  Saved /tmp/test/idna-ssl-1.1.0.tar.gz
Collecting idna>=2.0 (from yarl<2.0,>=1.0->aiohttp)
  Using cached https://files.pythonhosted.org/packages/29/88/c52aae187d3b128a0f13f36a6c987fc0d408d03a678ad9996516925d8495/idna-3.1-py3-none-any.whl
  Saved /tmp/test/idna-3.1-py3-none-any.whl
Successfully downloaded aiohttp attrs chardet multidict async-timeout yarl typing-extensions idna-ssl idna
```

And for package that missed in `python_deps`
```
 pip3 download rapid-router --dest=/tmp/test/
Collecting rapid-router
  Url 'file:///opt/seagate/cortx/upgrades/2.0.0-757/python_deps/rapid-router/' is ignored: it is neither a file nor a directory.
  Using cached https://files.pythonhosted.org/packages/fa/01/8ed102f11d7630883ab812fbbe414811834417bd3a2f480a1c4150bcad60/rapid_router-2.6.1-py2.py3-none-any.whl
  Saved /tmp/test/rapid_router-2.6.1-py2.py3-none-any.whl
Collecting cfl-common (from rapid-router)
  Url 'file:///opt/seagate/cortx/upgrades/2.0.0-757/python_deps/cfl-common/' is ignored: it is neither a file nor a directory.
  Using cached https://files.pythonhosted.org/packages/a3/d4/6c197cc9d33c6e424023cd1087f88a852cda6a245e8089f2c04fda250358/cfl_common-4.28.4-py3-none-any.whl
  Saved /tmp/test/cfl_common-4.28.4-py3-none-any.whl
Collecting djangorestframework==3.12.2 (from rapid-router)
  Url 'file:///opt/seagate/cortx/upgrades/2.0.0-757/python_deps/djangorestframework/' is ignored: it is neither a file nor a directory.
  Using cached https://files.pythonhosted.org/packages/8e/42/4cd19938181a912150e55835109b1933be26b776f3d4fb186491968dc41d/djangorestframework-3.12.2-py3-none-any.whl
  Saved /tmp/test/djangorestframework-3.12.2-py3-none-any.whl
Collecting future==0.18.2 (from rapid-router)
  Url 'file:///opt/seagate/cortx/upgrades/2.0.0-757/python_deps/future/' is ignored: it is neither a file nor a directory.
  Using cached https://files.pythonhosted.org/packages/45/0b/38b06fd9b92dc2b68d58b75f900e97884c45bedd2ff83203d933cf5851c9/future-0.18.2.tar.gz
....

Successfully downloaded rapid-router cfl-common djangorestframework future libsass django-js-reverse django-foundation-statics more-itertools django-pipeline pyhamcrest django django-countries django-two-factor-auth wagtail six sqlparse pytz django-otp django-formtools qrcode django-phonenumber-field django-treebeard draftjs-exporter Pillow requests django-modelcluster html5lib anyascii l18n Willow beautifulsoup4 Unidecode tablib xlsxwriter django-filter django-taggit babel urllib3 chardet certifi idna webencodings soupsieve xlrd xlwt openpyxl et-xmlfile
```